frontends/p4/specializeGenericTypes.h

```diff
@@ -166,6 +166,27 @@ class SpecializeGenericTypes : public PassRepeated {
         passes.emplace_back(new ClearTypeMap(typeMap, true));
         specMap.refMap = refMap;
         specMap.typeMap = typeMap;
+        setName("DoSpecializeGenericTypes");
+    }
+};
+
+class RemoveGenericTypes : public Transform {
+ public:
+    const IR::Node* postorder(IR::Type_StructLike* type) override {
+        if (!type->typeParameters->empty()) return nullptr;
```

Comment:
There is a nullpointer dereference according to the tests. I am guessing because the type is now null because of this?

Reply:
In fact this is a bug in the type checker which does not handle some expressions. I will add a commit to this PR to fix that.